independent-projects/qute/core/src/main/java/io/quarkus/qute/Locate.java

```diff
@@ -0,0 +1,40 @@
+package io.quarkus.qute;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * A custom {@link TemplateLocator}s are not available during the build time, therefore {@link Template} located by
+ * the locator must disable its validation by annotating the {@link TemplateLocator} with this annotation. If
+ * {@link TemplateLocator} locates {@link Template} and fails to declare the fact this way, an exception is thrown and the build
+ * fails.
+ *
+ * An example:
+ *
+ * <pre>
+ * &#64;Locate("/my/custom/location")
+ * public class MyCustomLocator implements TemplateLocator {
+ *
+ *     &#64;Override
+ *     public Optional<TemplateLocation> locate(String s) {
+ *         return Optional.empty();
+ *     }
+ *
+ * }
+ * </pre>
+ */
+@Target(ElementType.TYPE)
+@Retention(RUNTIME)
+@Repeatable(Locates.class)
+public @interface Locate {
+
+    /**
+     * @return template id
```

Comment:
I wonder if this should be a regular expression instead? I can imagine a custom template locator that is used for a large group of templates, and potentially the names of the templates are not known beforehand but instead have a specific prefix...

Reply:
Good idea, thank you.